### ISSUE TITLE
Grain Persistence and GrainCancellationTokenSource

### ISSUE THREAD

**Issue by TulkasLaugh** [dotnet/orleans#5299]
Hi, I am creating grain persistence for a master grain that manages the work of other worker grains. I am utilizing GrainCancellationTokenSource so the master grain can cancel all of the worker grains, however when I try to use Grain Persistence for the GrainCancellationTokenSource I get an error saying that GrainCancellationTokenSource cannot be deep copied. Do you guys have any recommendations around this error or have any recommendations for managing cancellation of worker grains? Thanks.

**Comment by sergeybykov** [comment]
`GrainCancellationTokenSource` is not meant to be persistent. It's a transient object that allows to try to cancel an in-flight requests. Why do you want to persist it when it has no meaning after the grain call it's used for completes?

**Comment by TulkasLaugh** [comment]
I was thinking that if a silo goes down in the middle of polling a job that's taking a long time and needs to be reactivated it would need the same CancellationTokenSource in order to still be able to cancel. Do you have a recommendation on trying to handle that kind of scenario?

**Comment by sergeybykov** [comment]
The general recommendation is not to make long running calls. Instead, make a call to *initiate* a long running operation, and get later notified (via a grain call, an observer or a stream) about its completion. If the grain that is executing the long running operation (on a separate thread or on the thread pool to avoid blocking grain threads) is marked as `[Reentrant]`, it can receive an explicit request to cancel the operation (without using the cancellation token) or get polled about the progress of the operation.

In other words, the recommended pattern is to make quick grain calls and manage any long running process via a series of them instead of a one long call. This pattern is more robust in the face of failures and in general makes the app logic more responsive and flexible.

**Comment by TulkasLaugh** [comment]
Make sense, I'll follow your recommendations.
